src/cobo/handshake.c

```diff
@@ -327,16 +344,65 @@ static int handshake_main(int sockfd, handshake_protocol_t *hdata, uint64_t sess
    /**
     * Recieve a packet_buffer on the network
     **/
+   result = recv_packet(sockfd, &recvd_long_packet_buffer, &recvd_long_packet_buffer_size);
+   if (result < 0) {
+      debug_printf("Problem receiving packet\n");
+      return_result = result;
+      socket_error = 1;
+      goto done;
+   }
+
+   /**
+    * Decrypt their packet with their random number
+   **/
+   debug_printf("Decrypting their random number packet\n");
```

Comment:
Misleading comment and printf. This is decrypting the main handshake packet (which does contain the initial random number). But it is not the random number packet, which this comment implies.